### DISCUSSION TITLE
Copy attachment image url to separate DB field

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#7157] [Help] [answered]
Hello

I'm currently building out a CSV export for my app and I want to include the url of the images I have in the attachment field of my DB. Is there anyway of accessing this image url either through an automation or through a formula field? I've looked through all previous discussions, but there aren't any solutions for accessing attachment array data in the backend. 

I could pull out the URL in the front end and then e.g. create some action chain to happen based on a button click, but I really would want to automate this process in the backend. 

Any bright ideas?

**Comment by R2bEEaton** [comment, marked as answer]
Hello there,

My recommendation would be to save the image URL when the attachment is uploaded. So let's say you have a button that uploads the attachment. Add another action after the Save Row which will save the row again and add the attachment url to another column.
![image](https://user-images.githubusercontent.com/34921506/183499410-b13aca19-5785-466c-87bc-182af75b16f8.png)

Using... `{{ Action 1.Saved row.image.0.url }}`.

Attached is an export of an example of this for you to investigate further. Please note you will need to append your domain to each image URL. You can either do this with handlebars upon upload or manually after exporting depending on your use case.
[Test-export-1659987309936.txt](https://github.com/Budibase/budibase/files/9284531/Test-export-1659987309936.txt)

**Reply by ghost** [reply]
Hi @R2bEEaton 

Thanks for the suggestion. I think this might work.

The only issue I'm running into the moment is that the image is failing to upload. This is the error I'm getting. I've tried to find solutions on here (such as #5697) but nothing seems to work for me. 
<img width="1184" alt="Screenshot 2022-08-09 at 09 04 55" src="https://user-images.githubusercontent.com/64409787/183597788-4285abe1-cd83-43c1-83d6-2c9777d2fb9d.png">

This is how I've set up the attachment field:
<img width="262" alt="image" src="https://user-images.githubusercontent.com/64409787/183599196-e9fd01ce-258f-4b60-baf1-bf16fcf5734f.png">

And this is how I've set up the button actions to save the row. I have a text field above the attachment field that I want the user to enter a e.g. unique ID so the attachment is uploaded to a specific record in my DB.
<img width="1569" alt="Screenshot 2022-08-09 at 09 15 57" src="https://user-images.githubusercontent.com/64409787/183599692-2b17792f-01a0-408e-8998-ce79621d97e7.png">

Have I gone wrong somewhere? The info in the Budibase docs about attachment fields doesn't really explain in detail about how to use the attachment component. It only says (and I'm paraphrasing) 'add it to your form and it should work right away'.

**Reply by R2bEEaton** [reply]
Yes, I noticed this bug when I was testing solutions for your question. Until it gets fixed, either use JavaScript like it mentions in the discussion post you linked, OR simply remove the image field from your first "Save Row." Since you selected your form as the data source, it will automatically take from there. Just make sure the attachment field name is the same as the column you want it to be in your table. You can also remove the ID field similarly if you want.

"Choosing a data source will automatically use the data it provides."

I'd also recommend limiting the number of attachments to 1 in the attachment element properties unless you modify the handlebars I gave for getting the URL a bit. Right now it just gets the first URL.

**Reply by ghost** [reply]
Ok, I will try that and let you know if I run into any other problems. 

Is there anyway we can flag this bug again?

Thanks!

**Reply by R2bEEaton** [reply]
Ok, sounds good!

> And this is how I've set up the button actions to save the row. I have a text field above the attachment field that I want the user to enter a e.g. unique ID so the attachment is uploaded to a specific record in my DB.

If you didn't already know this, if you want it to replace an existing row you'll have to pass that row's `_id`. If you use your own `ID` field, it will just make a new row. https://docs.budibase.com/docs/actions#save-row

As for the bug, I'll send a link to your this post in the comments.

**Reply by ghost** [reply]
Ok, great. Thanks for the clarification on the rows `_id`. That's really helpful!

**Reply by R2bEEaton** [reply]
@adamjdavies Sure thing. Did this work for you? If so, please go ahead and mark your question as answered so it can help people in the future 👍.
